CONTRIBUTING.md

```diff
@@ -73,23 +73,26 @@ The workflow is as follows for a _new_ protocol:
     ![](src/management/pr-on-github-4.png)
 
 1. If the reviewers raise concerns, changes can be made to the protocol that address these concerns (stage, commit, push)
-1. When all reviewers have given their approval, the repo admin:
-    1. adds tags with definitive version numbers to the YAML header 
-    1. updates the repo NEWS.md file
-    1. and merges the PR to the master
+1. When all reviewers have given their approval, the **repo admin** needs to do some necessary admin tasks before merging [see RELEASES.md](RELEASES.md)
 1. The GitHub protocols repo is setup in such a way that branches that are merged in the master branch will be deleted automatically.
 
 
-For an _update_ of an existing protocol all steps are the same, except for:
+For an **update** of an existing protocol all steps are the same, except for:
```

Comment:
A further idea for the protocolhelper package? A function that creates and checks out a new branch (protocol-specific), branching off from origin/master (even if one has not checked out master) after fetching origing/master. And also it adds ".dev" to the version number. No auto-committing though, but perhaps staging of the latter. It would just need the protocol code as an input.

Something like:

```r
protocol_code <- "string-by-user" # check if it exists
```

And then this git workflow, wrapped in `system()` or mimicked by `git2r`:

```
git fetch origin master  # 'origin' could also be generalized to take the first line of the output of 'git remote'
git branch protocol_code origin/master # make the branch
git push -u origin protocol_code:protocol_code # push the branch
git checkout protocol_code # to be put at the end because it is not a fail-safe command (it may error because of unclean working directory)
```

Plus the adding of `.dev` (without committing, but perhaps already staged (`git add -u`)).

Reply:
idemdito  see inbo/protocolhelper#27

Reply:
The workflow now mentions the use of `checklist::create_branch()` (currently only in the renv-setup branch) which deals with most steps (except the final step) listed here. So I think we can just use that, possibly wrapped into a protocolhelper function to include the final step.

Reply:
Sounds like a good idea!

Reply:
@ThierryO after checking the code of `checklist::new_branch` I noticed that https-protocol to talk to GitHub is not supported. So I guess only SSH is supported. That asks quite a lot of confusing [setup-steps](https://happygitwithr.com/ssh-keys.html) from often rather inexperienced Rusers who want to contribute to the protocolsource repo. Is this design decision fixed or can https be allowed. I can write an issue or a PR if I understand the code.

Reply:
ssh is not that hard to setup, if you know what to do. And you have to do it only once on your machine. We could ask the IT department to do that when people get a new machine. And we can do it with people need it on their current machine.

Reply:
OK, can you share what steps I (we) have to take? I checked the checklist documentation, but did not immediately found it (might have missed it).

Reply:
1. https://happygitwithr.com/ssh-keys.html#option-1-set-up-from-rstudio
2. https://happygitwithr.com/ssh-keys.html#windows-1
3. https://happygitwithr.com/ssh-keys.html#rstudio-to-clipboard
4. https://happygitwithr.com/ssh-keys.html#on-github

Reply:
After [reading this](https://github.com/r-lib/gert#should-i-use-https-or-ssh-remotes), it leaves me doubting a little about the preference for SSH.

Reply:
Several NARA colleagues are working with ssh for the latest two years now. So far, only one intervention needed. Due to somebody with a new computer.

Reply:
I followed the steps (and even wrote a vignette for checklist with all steps), but when I try `checklist::new_branch()` (within my local clone of the checklist package I want to make a branch to commit the vignette), it fails with this error:

```
Error in fetch(repo, "origin", verbose = verbose) : 
  Error in 'git2r_remote_fetch': error authenticating: failed connecting agent 
```

This happens inside `checklist::clean_git()`. I have set `git remote set-url origin git@github.com:inbo/checklist.git`.

I have no clue what I did wrong. On the other hand, I think issues such as these can be avoided by switching `git2r` functions to the equivalent `gert` functions which has [automatic authentication](https://github.com/r-lib/gert#automatic-authentication) (and which would also allow both SSH and HTTPS).

Reply:
Switching for `git2r` to `gert` is a major change. Something I cannot do on short notice. I'm willing to accept a PR .

Reply:
I might give it a try, but it will also take some time...